src/library/banshee/bansheeplaylistmodel.cpp

```diff
@@ -31,39 +32,50 @@
 #define CLM_COMPOSER "composer"
 #define CLM_PREVIEW "preview"
 
+namespace {
+QAtomicInt sTableNumber;
+}
+
 BansheePlaylistModel::BansheePlaylistModel(QObject* pParent, TrackCollection* pTrackCollection, BansheeDbConnection* pConnection)
         : BaseSqlTableModel(pParent, pTrackCollection, "mixxx.db.model.banshee_playlist"),
           m_pConnection(pConnection),
           m_playlistId(-1) {
+    m_tempTableName = BANSHEE_TABLE + QString::number(sTableNumber.fetchAndAddAcquire(1));
 }
 
 BansheePlaylistModel::~BansheePlaylistModel() {
+    deleteTempTable();
 }
 
-void BansheePlaylistModel::setTableModel(int playlistId) {
-    //qDebug() << "BansheePlaylistModel::setTableModel" << playlistId;
-    if (m_playlistId == playlistId) {
-        qDebug() << "Already focused on playlist " << playlistId;
-        return;
-    }
-
+void BansheePlaylistModel::deleteTempTable() {
     if (m_playlistId >= 0) {
         // Clear old playlist
         m_playlistId = -1;
         QSqlQuery query(m_pTrackCollection->database());
-        QString strQuery("DELETE FROM " BANSHEE_TABLE);
-        if (!query.exec(strQuery)) {
+        QString strQuery("DELETE FROM %1");
```

Comment:
The empty table remains. Please drop it entirely instead of only deleting all records:
`DROP TABLE IF EXISTS %1`
https://www.sqlite.org/lang_droptable.html